src/Symfony/Bundle/FrameworkBundle/Controller/AbstractController.php

```diff
@@ -64,7 +65,9 @@ public function setContainer(ContainerInterface $container)
     protected function getParameter(string $name)
     {
         if (!$this->container->has('parameter_bag')) {
-            throw new \LogicException('The "parameter_bag" service is not available. Try running "composer require dependency-injection:^4.1"');
+            throw new ServiceNotFoundException('parameter_bag', null, null, array(),
+                'The "parameter_bag" service could not be located. Ensure that the controller is either set '.
+                'to be autoconfigured or wired manually to inject it.');
```

Comment:
`[...] to be autoconfigured or has the "container.service_subscriber" tag.`?

should be on one single long line

Reply:
`sprintf('The "%s::getParameter()" method is missing a parameter bag to work properly. Did you forget to register your controller as a service subscriber? This can be done e.g. by using autoconfiguration or by manually wiring a "parameter_bag" in the service locator passed to the controller.', get_class($this));

(the name of the parameter is of no importance IMHO, but the controller class IS.)